cmd/dagger/call.go

```diff
@@ -55,6 +44,7 @@ dagger call lint stdout
 		case Terminal:
 			c.Select("websocketEndpoint")
 		default:
+			// FIXME: don't make this an error, it's confusing to users
 			return fmt.Errorf("return type %q requires a sub-command", name)
```

Comment:
If not for this you'll get an error from GraphQL anyway. It was added to provide a better error message. 

It will be shown less often when we print scalars from objects by default but there's some technical pre-requisites to allow for that.

This error still needs to exist though, since it's possible you have an object without scalars to print, if it only acts as a constructor to other objects to chain, for example. And if you return a `CacheVolume` for some reason, it only has an `id` field and I don't think we'll print ids as part of the feature mentioned above.

If the error is confusing to users, do you have a better alternative besides improving the error message?

Reply:
It's just a general note of a problem (calling a function that returns an object without error, should not return an error to the user). I don't have a specific solution at the moment (tried initially in this PR, but gave up). Hence the FIXME.

Reply:
To make a query you need to end in a scalar/leaf, so this error needs to exist. We can only reduce the chance of hitting it when we can print an object's fields but that won't replace this error 100% of the time.

Reply:
Those are artificial constraints. If we decide we want to replace the error 100% of the time, it's easy to do so. The question is do we agree on doing it.

Example: if the object has no scalars, print nothing. Or, print the ID. Or, print the same message as before, but without prefixing it with "error". All these are plausible solutions to the problem.

Reply:
Yeah, I guess what I'm saying is that returning nothing or returning an ID doesn't seem better to me but let's discuss at another time.